group/views.py

```diff
@@ -20,3 +23,100 @@ def create_group(request):
             return Response(serializer.data, status=status.HTTP_201_CREATED)
         return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
     return None
+
+
+@api_view(['PATCH'])
+@permission_classes([IsAuthenticated])
+def add_users(request):
+    """Add user(s) to a group"""
+    if request.method == 'PATCH':
+        data = request.data.copy()
+        group_id = request.data.get("group_id")
+
+        data.pop('creator', None)
+        data.pop('name', None)
+        try:
+            group = Group.objects.get(id=group_id)
+            if request.user != group.creator:
+                return Response({
+                                "error": "Only group creator can add members"},
+                                status=status.HTTP_403_FORBIDDEN)
+            errors = []
+            for member in group.members.all():
+                if member.id not in data['members']:
+                    data['members'].append(member.id)
+            for member in data['members']:
+                try:
+                    user = User.objects.get(id=member)
+                    if user.id not in group.members.all()\
```

Comment:
What you are doing here is enough to check whether a user already exists in the group! So the for loop on line 45 is unnecessary

Reply:
I think the confusion is coming from her using the copied dictionary from the request parameter. Basically avoid modifying user supplied data unless you have to. e.g if a user supplied members as part of their data, do you realise how your logic would be dependant on what they added in that data? They can actually circumvent your checks and it's a serious security issue. Start from a fresh variable with no an empty array. And as Thatcher said, this can happen in one loop instead of two.